HW2: Markdown file
================
Kevin Liu
2023-09-30

# Problem 1

## First, clean the data in pols-month.csv.

Use separate() to break up the variable mon into integer variables year,
month, and day; replace month number with month name; create a president
variable taking values gop and dem, and remove prez_dem and prez_gop;
and remove the day variable.

``` r
pols_month_df = 
  read_csv("./data/pols-month.csv") |>
  janitor::clean_names() |> 
  separate(mon, into = c("year", "month", "day"), convert = TRUE) |> 
  mutate(month = month.abb[month]) |> 
  
  #This leaves leading zeros, which int will delete... but should be fine...
  mutate(year = substr(year,2,3)) |>  
  mutate(year = as.integer(year)) |> 
#Redundant code...I should have used "convert"!
  # separate(mon, into = c("year", "month_day"), sep = 4) |> 
  # mutate(year = as.numeric(year)) |> 
  # separate(month_day, into = c("month", "day"), sep = 3) |> 
  # mutate(month = substr(month, 2,3)) |> 
  # mutate(month = as.numeric(month)) |> 
  # mutate(month = month.name[month]) |> 
  # mutate(day = substr(day, 2,3)) |> 
  # mutate(day = as.numeric(day)) |> 

  mutate(president = if_else(prez_gop == 1 | prez_gop == 2, "gop", "dem")) |> 
  select(year, everything()) |> 
  select(-prez_gop, -prez_dem, -day)
```

    ## Rows: 822 Columns: 9
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: ","
    ## dbl  (8): prez_gop, gov_gop, sen_gop, rep_gop, prez_dem, gov_dem, sen_dem, r...
    ## date (1): mon
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
#pols_month_df
```

## Second, clean the data in snp.csv using a similar process to the above.

For consistency across datasets, arrange according to year and month,
and organize so that year and month are the leading columns.

``` r
snp_df = 
  read_csv("./data/snp.csv") |> 
  janitor::clean_names() |> 
  separate(date, into = c("month", "day", "year"), convert = TRUE) |> 
  mutate(month = month.abb[month]) |>
  select(-day) |> 
  select(year,everything())
```

    ## Rows: 787 Columns: 2
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: ","
    ## chr (1): date
    ## dbl (1): close
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
#The way i did is not robust for the given date
  # separate(date, into = c("month", "day_year"), sep = 1) |> 
  # mutate(month = as.numeric(month)) |> 
  # mutate(month = month.name[month]) |> 
  # separate(day_year, into = c("day", "year"), sep = 2) |> 
  # select(-day) |> 
  # mutate(year = substr(year, 2,3)) |> 
  # mutate(year = if_else(grepl("/", year), substr(year,2,3), year))
  # mutate(year = as.numeric(year)) |> 
  # arrange(year)
#snp_df
```

## Third, tidy the unemployment data so that it can be merged with the previous datasets.

This process will involve switching from “wide” to “long” format;
ensuring that key variables have the same name; and ensuring that key
variables take the same values.

``` r
unemployment_df = 
  read_csv("./data/unemployment.csv") |> 
  pivot_longer(
    Jan:Dec,
    names_to = "month",
    values_to = "unemployment") |> 
  mutate(Year = substr(Year,3,4)) |> 
  rename("year" = "Year") |> 
  mutate(year = as.integer(year))
```

    ## Rows: 68 Columns: 13
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: ","
    ## dbl (13): Year, Jan, Feb, Mar, Apr, May, Jun, Jul, Aug, Sep, Oct, Nov, Dec
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

## Join the datasets by merging snp into pols, and merging unemployment into the result.

``` r
polls_snp_df = 
  left_join(pols_month_df, snp_df)
```

    ## Joining with `by = join_by(year, month)`

``` r
polls_snp_unem_df =
  left_join(polls_snp_df, unemployment_df)
```

    ## Joining with `by = join_by(year, month)`

### Write a short paragraph about these datasets.

Explain briefly what each dataset contained, and describe the resulting
dataset (e.g. give the dimension, range of years, and names of key
variables).

**Response** The “pols” dataset contains 822 observations of 9 variables
related to the number of national politicians who are democratic or
republican at any given time from 1947 to 2015. Of the 9 variables, 1 is
the date, 4 variables enumerate how many presidents, govenors, senators,
and representatives that are democrats, and the last 4 is the same
enumeration but for repbulicans. The “snp” dataset contains 787
observations of 3 variables related to Standard & Poor’s stock market
index (S&P). It contains a date variable and the S&P’s closing value on
that date from 1950 to 2015. The “unemployment” dataset contains 816
observations of 3 variables. The first column variable the year, denotes
the year for the row while the next 12 column variables designates the
month of that year. Within the cells are the unemployment rates for a
partcular month in a year. During data processing, “unemployement” was
long-ified and contains data from 1948 to 2015. The resulting dataset
has a combination of all three datasets. “Unemployment” has been made
long-ified in order to be properly merged with “pols” and “snp” based on
month and year. It is a dataset with 822 observations and 11 variables
where all three were merged based on year + month with close +
unemployment being added to “pols” dataset.

# Problem 2

## Read and clean the Mr. Trash Wheel sheet:

- specify the sheet in the Excel file and to omit non-data entries (rows
  with notes / figures; columns containing notes) using arguments in
  read_excel
- use reasonable variable names
- omit rows that do not include dumpster-specific data

``` r
mr_trash_df =
  readxl::read_xlsx("./data/202309 Trash Wheel Collection Data.xlsx", sheet = "Mr. Trash Wheel", skip = 1, n_max = 584) |> 
  janitor::clean_names() |> 
  mutate(dump_name = "mstr") |> 
  select(-x15, -x16) |> 
  select(dump_name, everything()) |> 
  mutate(year = as.double(year))
```

    ## New names:
    ## • `` -> `...15`
    ## • `` -> `...16`

## Update the data to include a new homes_powered variable based on this calculation.

The data includes a column for the (approximate) number of homes
powered. This calculation is described in the Homes powered note, but
not applied to every row in the dataset.

- Homes Powered - Each ton of trash equates to on average 500 kilowatts
  of electricity. An average household will use 30 kilowatts per day.”

``` r
mr_trash_df = 
  mutate(mr_trash_df, homes_powered = (weight_tons * 500) / 30)
```

## Use a similar process to import, clean, and organize the data for Professor Trash Wheel and Gwynnda..

and combine these with the Mr. Trash Wheel dataset to produce a single
tidy dataset. To keep track of which Trash Wheel is which, you may need
to add an additional variable to all datasets before combining.

### Professor Trash Wheel

``` r
prof_trash_df =
  readxl::read_xlsx("./data/202309 Trash Wheel Collection Data.xlsx", sheet = "Professor Trash Wheel", skip = 1, n_max = 106) |> 
  janitor::clean_names() |> 
  mutate(dump_name = "prof") |> 
  select(dump_name, everything())
```

### Gwynnda Trash Wheel

``` r
gwynnda_trash_df =
  readxl::read_xlsx("./data/202309 Trash Wheel Collection Data.xlsx", sheet = "Gwynnda Trash Wheel", skip = 1, n_max = 155) |> 
  janitor::clean_names() |> 
  mutate(dump_name = "gwyn") |> 
  select(dump_name, everything())
```

### Merge All 3 Trash Wheels

``` r
mstr_prof_gwynnda_df = 
  bind_rows(mr_trash_df, prof_trash_df, gwynnda_trash_df)
```

## Write a paragraph about these data; you are encouraged to use inline R.

Be sure to note the number of observations in the resulting dataset, and
give examples of key variables.

**Response** mr_trash_df contains the converted xlsx file to R studio
dataframe data which has 584 observations of which are the days that the
trash wheel was deployed and used. The amount of trash collected was
enumerated (ie. weight_tons) and the types of trash collected where
enumerated as well (ie. plastic_bottles) and a final “houses_powered”
variable results as a way to measure the amount of trash collected.
prof_trash_df and gwynna_trash_df are similar but contain 106
observations and 155 observations respectively. The final combination of
the 3 contains the sum of the three separate dataframes as the
dataframes are stacked together into one dataframe.

### For available data, what was the total weight of trash collected by Professor Trash Wheel?

**Response** The total weight of trash collected by professor trash
wheel was 1875.1 tons.

### What was the total number of cigarette butts collected by Gwynnda in July of 2021?

``` r
gwynnda_trash_df_1 =
  filter(gwynnda_trash_df, month == "July") |> 
  filter(year == 2021)
```

**Response** The total \# of cigarette butts collected by Gwynnda in
July of 2021 is 1.63^{4} butts.

# Problem 3

This problem uses data collected in an observational study to understand
the trajectory of Alzheimer’s disease (AD) biomarkers. Study
participants were free of Mild Cognitive Impairment (MCI), a stage
between the expected cognitive decline of normal aging and the more
serious decline of dementia, at the study baseline.

Basic demographic information were measured at the study baseline. The
study monitored the development of MCI and recorded the age of MCI onset
during the follow-up period, with the last visit marking the end of
follow-up. APOE4 is a variant of the apolipoprotein E gene,
significantly associated with a higher risk of developing Alzheimer’s
disease. The amyloid β42/40 ratio holds significant promise for
diagnosing and predicting disease outcomes. This ratio undergoes changes
over time and has been linked to the manifestation of clinical symptoms
of Alzheimer’s disease.

## Import, clean, and tidy the dataset of baseline demographics.

Ensure that sex and APOE4 carrier status are appropriate encoded
(i.e. not numeric), and remove any participants who do not meet the
stated inclusion criteria (i.e. no MCI at baseline). Discuss important
steps in the import process and relevant features of the dataset.

``` r
mci_baseline_df = 
  
  #read_csv: The original data has two rows that contain variable information. The first one is skipped in order for the column variable names to have the correct column name but still contains important information! 
  read_csv("./data/MCI_baseline.csv", skip = 1) |>
  
  #janitor::clean_names(): makes sure all variables are consistently camel case!
  janitor::clean_names() |> 
  
  #recode: code taken/inspired from/by (thanks!): https://www.r-bloggers.com/2022/06/how-to-recode-values-in-r/
  #mutate: The first row in the .csv file imported had indicated that female = 0 and male = 1.
  mutate(sex = recode(sex, '0'='Female', '1'='Male'))  |> 
  mutate(apoe4 = recode(apoe4, '0'='APOE4 Non-Carrier', '1'='APOE4 Carrier')) |> 
  
  #filter: includes all rows that do NOT have "." or missing data. This was indicated by the first row of the imported .csv where it had described that those with "."/missing age are those that were MCI free and so these are being filtered out.
  filter(age_at_onset != ".") |> 
  filter(age_at_onset > current_age) 
```

    ## Rows: 483 Columns: 6
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: ","
    ## chr (1): Age at onset
    ## dbl (5): ID, Current Age, Sex, Education, apoe4
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

## How many participants were recruited, and of these how many develop MCI?

Based on the initial import of the dataset (after skipping the first
row), the resulting initial dataframe contains 483 observations and thus
483 participants were initially recruited. However, there are subjects
that have been diagnosed with MCI before their current age??? (The
question problem states that the study recruited MCI free patients????)
And so these are removed which resulted in 479 observations. After the
complete processing of the dataframe (after recoding sex and filtering
out those with missing ages), the dataframe contained 93 observations
and thus 93 participants developed MIC.

## What is the average baseline age?

``` r
#The "mci_baseline_prefilter_df" is the same as "mci_baseline_df" but before filtering out missing onset age. 
mci_baseline_prefilter_df = 
  read_csv("./data/MCI_baseline.csv", skip = 1) |>
  janitor::clean_names() |> 
  mutate(sex = recode(sex, '0'='Female', '1'='Male'))  |> 
  mutate(apoe4 = recode(apoe4, '0'='APOE4 Non-Carrier', '1'='APOE4 Carrier')) |> 
  filter(age_at_onset > current_age | age_at_onset == ".")
```

    ## Rows: 483 Columns: 6
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: ","
    ## chr (1): Age at onset
    ## dbl (5): ID, Current Age, Sex, Education, apoe4
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
mean(pull(mci_baseline_prefilter_df, current_age))
```

    ## [1] 65.0286

The average baseline age, which presumably means all participants that
were recruited at baseline and still includes those that did not develop
MCI as well as those that did devlope MCI, is 65.0286 years old.

## What proportion of women in the study are APOE4 carriers?

``` r
#create intermediate dataset, for the purpose of the assignment, include when sex = 0, female. Additionally in order to properly calculate the propotion, we remove the initial label given to the apoe4 variable in the initial dataset.
mci_baseline_women_df =
  read_csv("./data/MCI_baseline.csv", skip = 1) |>
  janitor::clean_names() |> 
  mutate(sex = recode(sex, '0'='Female', '1'='Male'))  |> 
  
  #The below is removed to take the mean of apoe4 which should produce a proportion. If 1 = APOE4 carrier and 0 = APOE4 noncarrier, the resulting mean should be a proportion.
  
  #mutate(apoe4 = recode(apoe4, '0'='APOE4 Non-Carrier', '1'='APOE4 Carrier')) |> 
  filter(age_at_onset > current_age | age_at_onset == ".") |> 
  filter(sex == "Female") 
```

    ## Rows: 483 Columns: 6
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: ","
    ## chr (1): Age at onset
    ## dbl (5): ID, Current Age, Sex, Education, apoe4
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

``` r
mean(pull(mci_baseline_women_df, apoe4))
```

    ## [1] 0.3

The proportion of women in the study that are APOE4 carriers is 0.3 or
30% of the women in the study.

## Similarly, import, clean, and tidy the dataset of longitudinally observed biomarker values;

…comment on the steps on the import process and the features of the
dataset.

``` r
mci_amyloid_df = 
  
  #read_csv: The original data has two rows that contain variable information. The first one is skipped in order for the column variable names to have the correct column name.
  #NOTE: the columns are the amyloid b42/40 ratios at different time windows
  read_csv("./data/mci_amyloid.csv", skip = 1) |>
  
  #janitor::clean_names(): makes sure all variables are consistently camel case!
  janitor::clean_names() |> 

  #pivot_longer: TIDY DATA! 
  pivot_longer(
    baseline:time_8,
    names_to = "test_time", 
    values_to = "amyloid_ratio") 
```

    ## Rows: 487 Columns: 6
    ## ── Column specification ────────────────────────────────────────────────────────
    ## Delimiter: ","
    ## chr (5): Baseline, Time 2, Time 4, Time 6, Time 8
    ## dbl (1): Study ID
    ## 
    ## ℹ Use `spec()` to retrieve the full column specification for this data.
    ## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.

## Check whether some participants appear in only the baseline or amyloid datasets,

…and comment on your findings.

``` r
#I decided to adjust column variable name in mci_amyloid from "study_id" to "id" so it is joinable.
mci_amyloid_join_df = 
  mci_amyloid_df |> 
  setNames(c("id", "test_time", "amyloid_ratio"))

#This dataframe contains data of all processed baseline before filtering out missing.
mci_baseline_join_df =
  mci_baseline_prefilter_df

#Full join will help with determining which variables are missing. If all variables from the baseline is missing for a certain row, then that participant is missing in the baseline. If all variables from the amyloid biomarker csv is missing, then that partiicpant is missing in the amyloid dataset but present in the baseline dataset.  
mci_join = full_join(mci_baseline_join_df, mci_amyloid_join_df, by = "id")
```

``` r
#To check if missing from baseline dataset, filter out participant data that contain non-missing data in baseline columns, Current_Age:Age_at_onset.
mci_baseline_missing =
  filter(mci_join, is.na(current_age) &
           is.na(sex) &
           is.na(education) &
           is.na(apoe4) &
           is.na(age_at_onset))

#The resulting dataset should provide a list of IDs that are present in amyloid dataset but missing from baseline dataset

#
```

Result: There are 16 participant ids that had missing data in all
columns in the baseline data. Subjects 72, 234, 283, 380, 484 - 495

``` r
#To check if missing from baseline dataset, filter out participant data that contain non-missing data in amyloid columns, baseline:time_8
mci_amyloid_missing =
  filter(mci_join, 
         is.na(test_time) &
           is.na(amyloid_ratio))

#The resulting dataset should provide a list of IDs that are present in baseline dataset but missing from amyloid dataset
```

Result: There are 8 participant ids that had missing data in all columns
in the amyloid data. Subjects 14, 49, 92, 179, 268, 304, 389, 412

## Combine the demographic and biomarker datasets so that only participants who appear in both datasets…

…are retained, and briefly describe the resulting dataset; export the
result as a CSV to your data directory.

``` r
mci_inner_join = 
  inner_join(mci_baseline_join_df, mci_amyloid_join_df) 
```

    ## Joining with `by = join_by(id)`

``` r
# mci_test =
#   filter(mci_inner_join, is.na(test_time) &
#            is.na(amyloid_ratio) &
#            is.na(current_age) &
#            is.na(sex) &
#            is.na(education) &
#            is.na(apoe4) &
#            is.na(age_at_onset))

write.csv(mci_inner_join, "./data/MCI_join.csv", row.names=FALSE)
```

Result: The resulting data base the is the tidied version of a inner
join between amyloid dataset and baseline dataset. The dataset has 2355
observations where there are 471 participants
